internals/daemon/api_pairing.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2024 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package daemon
+
+import (
+	"encoding/json"
+	"net/http"
+)
+
+func v1PostPairing(c *Command, r *http.Request, user *UserState) Response {
+	var payload struct {
+		Action string `json:"action"`
+	}
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&payload); err != nil {
+		return BadRequest("cannot decode request body: %v", err)
+	}
+
+	switch payload.Action {
+	case "pair":
+		switch transport := RequestTransportType(r); transport {
+		case TransportTypeHTTPS:
```

Comment:
This is already checked in the PairingAccess access checker, right? (And it would respond with Unauthorized.) In that case, I'd prefer to do a sanity check at the top of this function and return an InternalError. Something like:

```go
func v1PostPairing(c *Command, r *http.Request, user *UserState) Response {
    if transport := RequestTransportType(r); transport != TransportTypeHTTPS {
        return InternalError("cannot pair using transport type %q", transport)
    }
    ...
```

Reply:
Yes, this feels more explicit. I also like that this is not an internal error, since we cannot actually make this happen if the access permission phase did its job. Fixed, thank you.

Reply:
Resolving.